tools/stern/src/Stern/Intra.hs

```diff
@@ -351,23 +351,22 @@ setBlacklistStatus status emailOrPhone = do
     statusToMethod False = DELETE
     statusToMethod True  = POST
 
-getLegalholdStatus :: TeamId -> Handler Bool
+getLegalholdStatus :: TeamId -> Handler SetLegalHoldStatus
 getLegalholdStatus tid = do
     info $ msg "Getting legalhold status"
     gly <- view galley
     (>>= fromResponseBody) . catchRpcErrors $ rpc' "galley" gly
          ( method GET
          . paths ["/i/teams", toByteString' tid, "features", "legalhold"]
-         . expect2xx
          )
   where
-    fromResponseBody :: Response (Maybe LByteString) -> Handler Bool
+    fromResponseBody :: Response (Maybe LByteString) -> Handler SetLegalHoldStatus
     fromResponseBody resp = case responseJsonEither resp of
-      Right (LegalHoldTeamConfig LegalHoldDisabled) -> pure False
-      Right (LegalHoldTeamConfig LegalHoldEnabled) -> pure True
+      Right (LegalHoldTeamConfig LegalHoldDisabled) -> pure SetLegalHoldDisabled
+      Right (LegalHoldTeamConfig LegalHoldEnabled) -> pure SetLegalHoldEnabled
       Left errmsg -> throwE (Error status502 "bad-upstream" ("bad response; error message: " <> pack errmsg))
```

Comment:
This is an error internal to the backend service family in the sense that the galley end-point always responds with a body that can be parsed as `LegalHoldTeamConfig`.  If it doesn't, this constitutes an invalid response from an upstream server.  502 is the correct status code for that.

If you read this, why do you expect galley to legitimately return anything I'm not handling?  And if you suspect that might be the case and want to be extra-thorough (which is great, since you're trying to find mistakes I made), why don't you go and check?

I think it is reasonable to expect the reader of the code to assume the author did the right thing.  What you are asking feels like the author should explicitly state for everything he does that he actually thought about it.  That's annoying for both author and reader.

Reply:
Okay, I made two comments on this PR, and the comment that there is a possible 403 that is not correctly handled is actually related to the `features/sso`, not `features/legalhold`, see comment below. The comment here is not applicable, as this endpoint seems to indeed always return a 200, even if the teamId passed in the request doesn't exist. That is another bug, though. (if galley were to properly refuse setting a feature flag for non-existent teams, galley would return a 4xx, and that 4xx would need to be handled in stern here, though - but that is a topic for a future PR).

Reply:
> I think it is reasonable to expect the reader of the code to assume the author did the right thing. What you are asking feels like the author should explicitly state for everything he does that he actually thought about it. That's annoying for both author and reader.

I admit I was lazy and asked a question before making sure if there is actually a problem. Sorry for that.